tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py

```diff
@@ -573,35 +631,89 @@ def test_calculate_weights_below_for_multi_objective() -> None:
     assert weights_below[1] == weights_below[2]
     assert sum(weights_below) > 0
 
+    # Zero/negative objective values.
+    weights_below = _tpe.sampler._calculate_weights_below_for_multi_objective(
+        {"x": np.array([1.0, 2.0, 3.0, 4.0], dtype=float)},
+        [(0, [-0.3, -0.3]), (0, [0.0, -0.8]), (0, [-0.8, 0.0]), (0, [1, 1])],
+        np.array([0, 1, 2]),
+    )
+    assert len(weights_below) == 3
+    assert weights_below[0] > weights_below[1]
+    assert weights_below[0] > weights_below[2]
+    assert np.isclose(weights_below[1], weights_below[2])
+    assert sum(weights_below) > 0
+
+    # +/-inf objective values.
+    weights_below = _tpe.sampler._calculate_weights_below_for_multi_objective(
+        {"x": np.array([1.0, 2.0, 3.0, 4.0], dtype=float)},
+        [(0, [-float("inf"), -float("inf")]),
+         (0, [0.0, -float("inf")]),
+         (0, [-float("inf"), 0.0]),
+         (0, [float("inf"), float("inf")])],
+        np.array([0, 1, 2]),
+    )
+    assert len(weights_below) == 3
+    assert all([np.isnan(w) for w in weights_below])
```

Comment:
[memo] Should not return `nan`'s list, but `[1, 0.5, 0.5, 0]` something like that. To be fixed.